openlabcmd/openlabcmd/service.py

```diff
@@ -26,27 +26,31 @@ class ServiceStatus(object):
     INITIALIZING = 'initializing'
     UP = 'up'
     DOWN = 'down'
-    Restarting = 'restarting'
+    RESTARTING = 'restarting'
+
+    @property
+    def all_status(self):
+        return [self.INITIALIZING, self.UP, self.DOWN, self.RESTARTING]
 
 
 class Service(object):
-    def __init__(self, name, node_role):
+    def __init__(self, name, node_role, alarmed=None, alarmed_at=None,
+                 restarted=None, restarted_at=None, is_necessary=None,
+                 status=None, **kwargs):
         self.name = name
-        self.alarmed = False
-        self.alarmed_at = None
-        self.updated_at = None
-        self.restarted = False
-        self.restarted_at = None
-        self.is_necessary = None
         self.node_role = node_role
-        self.status = ServiceStatus.INITIALIZING
+        self.alarmed = alarmed if alarmed else False
+        self.alarmed_at = alarmed_at if alarmed_at else None
+        self.restarted = restarted if restarted else False
+        self.restarted_at = restarted_at if restarted_at else None
+        self.is_necessary = is_necessary if is_necessary else None
```

Comment:
why need `... if ... else...` ? these vars already have the default `None` value

Reply:
+1